### PR TITLE
feat: GSM addresses

### PR DESCRIPTION
**Pull request by parth-15**

### CODE REVIEW

**Comment by parth-15** [comment]
close T-19100

File: scripts/configs/pools/ethereum.ts [hunk 1]
```diff
@@ -109,5 +109,9 @@ export const mainnetProtoV3Pool: PoolConfig = {
     WETH_GATEWAY: '0x893411580e590D62dDBca8a703d61Cc4A8c7b2b9',
     WITHDRAW_SWAP_ADAPTER: '0x78F8Bd884C3D738B74B420540659c82f392820e0',
     SAVINGS_DAI_TOKEN_WRAPPER: '0xE28E2c8d240dd5eBd0adcab86fbD79df7a052034',
+    GSM_USDC_FIXED_PRICE_STRATEGY: '0x430BEdcA5DfA6f94d1205Cb33AB4f008D0d9942a',
+    GSM_USDT_FIXED_PRICE_STRATEGY: '0x4c707764cbFB4FFa078e169e6b8A6AdbE7526a2c',
+    GSM_USDC_ORACLESWAPFREEZER: '0xef6beCa8D9543eC007bceA835aF768B58F730C1f',
+    GSM_USDT_ORACLESWAPFREEZER: '0x71381e6718b37C12155CB961Ca3D374A8BfFa0e5',
```
Comment (miguelmtzinf) [inline scripts/configs/pools/ethereum.ts]: These are not part of the Aave Pool. Move to Ethereum network

File: scripts/configs/networks/ethereum.ts [hunk 2]
```diff
@@ -32,7 +32,7 @@ export const ethereumAddresses: NetworkAddresses<{
     GHO_TOKEN: '0x40D16FC0246aD3160Ccc09B8D0D3A2cD28aE6C2f',
     GHO_FLASHMINTER_FACILITATOR: '0xb639D208Bcf0589D54FaC24E655C79EC529762B8',
     // TODO: perhaps makes sense to create a generator for GSM to automatically pull GSMs, for now let's see how this evolves
-    GSM_REGISTRY: '0x0d8eFfC11dF3F229AA1EA0509BC9DFa632A13578',
+    GSM_REGISTRY: '0x167527DB01325408696326e3580cd8e55D99Dc1A',
     GSM_USDC: '0x0d8eFfC11dF3F229AA1EA0509BC9DFa632A13578',
     GSM_USDT: '0x686F8D21520f4ecEc7ba577be08354F4d1EB8262',
   },
```
Comment (miguelmtzinf) [inline scripts/configs/networks/ethereum.ts]: Add fee strat

File: scripts/configs/pools/ethereum.ts [hunk 3]
```diff
@@ -109,5 +109,9 @@ export const mainnetProtoV3Pool: PoolConfig = {
     WETH_GATEWAY: '0x893411580e590D62dDBca8a703d61Cc4A8c7b2b9',
     WITHDRAW_SWAP_ADAPTER: '0x78F8Bd884C3D738B74B420540659c82f392820e0',
     SAVINGS_DAI_TOKEN_WRAPPER: '0xE28E2c8d240dd5eBd0adcab86fbD79df7a052034',
+    GSM_USDC_FIXED_PRICE_STRATEGY: '0x430BEdcA5DfA6f94d1205Cb33AB4f008D0d9942a',
+    GSM_USDT_FIXED_PRICE_STRATEGY: '0x4c707764cbFB4FFa078e169e6b8A6AdbE7526a2c',
+    GSM_USDC_ORACLESWAPFREEZER: '0xef6beCa8D9543eC007bceA835aF768B58F730C1f',
```
Comment (miguelmtzinf) [inline scripts/configs/pools/ethereum.ts]:
```suggestion
    GSM_USDC_ORACLE_SWAP_FREEZER: '0xef6beCa8D9543eC007bceA835aF768B58F730C1f',
```

File: scripts/configs/types.ts [hunk 4]
```diff
@@ -63,6 +63,10 @@ export interface PoolConfig {
     RATES_FACTORY?: Hex;
     PERMISSION_MANAGER?: Hex;
     WALLET_BALANCE_PROVIDER?: Hex;
+    GSM_USDC_FIXED_PRICE_STRATEGY?: Hex;
+    GSM_USDT_FIXED_PRICE_STRATEGY?: Hex;
+    GSM_USDC_ORACLESWAPFREEZER?: Hex;
+    GSM_USDT_ORACLESWAPFREEZER?: Hex;
```
Comment (miguelmtzinf) [inline scripts/configs/types.ts]: Move to `NetworkAddresses` type

File: scripts/configs/pools/ethereum.ts (hunk 1 above)
Comment (parth-15) [reply scripts/configs/pools/ethereum.ts]: Removed

File: scripts/configs/networks/ethereum.ts (hunk 2 above)
Comment (parth-15) [reply scripts/configs/networks/ethereum.ts]: added

File: scripts/configs/pools/ethereum.ts (hunk 3 above)
Comment (parth-15) [reply scripts/configs/pools/ethereum.ts]: done

File: scripts/configs/types.ts (hunk 4 above)
Comment (parth-15) [reply scripts/configs/types.ts]: done

File: scripts/configs/networks/ethereum.ts [hunk 5]
```diff
@@ -32,9 +32,14 @@ export const ethereumAddresses: NetworkAddresses<{
     GHO_TOKEN: '0x40D16FC0246aD3160Ccc09B8D0D3A2cD28aE6C2f',
     GHO_FLASHMINTER_FACILITATOR: '0xb639D208Bcf0589D54FaC24E655C79EC529762B8',
     // TODO: perhaps makes sense to create a generator for GSM to automatically pull GSMs, for now let's see how this evolves
-    GSM_REGISTRY: '0x0d8eFfC11dF3F229AA1EA0509BC9DFa632A13578',
+    GSM_REGISTRY: '0x167527DB01325408696326e3580cd8e55D99Dc1A',
     GSM_USDC: '0x0d8eFfC11dF3F229AA1EA0509BC9DFa632A13578',
     GSM_USDT: '0x686F8D21520f4ecEc7ba577be08354F4d1EB8262',
+    GSM_FIXED_FEE_STRATEGY: '0xD4478A76aCeA81D3768A0ACB6e38f25eEB6Eb1B5',
```
Comment (miguelmtzinf) [inline scripts/configs/networks/ethereum.ts]: let's add one per gsm

File: scripts/configs/networks/ethereum.ts [hunk 6]
```diff
@@ -32,9 +32,14 @@ export const ethereumAddresses: NetworkAddresses<{
...
+    GSM_USDC_FIXED_PRICE_STRATEGY: '0x430BEdcA5DfA6f94d1205Cb33AB4f008D0d9942a',
+    GSM_USDT_FIXED_PRICE_STRATEGY: '0x4c707764cbFB4FFa078e169e6b8A6AdbE7526a2c',
```
Comment (miguelmtzinf) [inline scripts/configs/networks/ethereum.ts]:
```suggestion
    GSM_USDC_PRICE_STRATEGY: '0x430BEdcA5DfA6f94d1205Cb33AB4f008D0d9942a',
    GSM_USDT_PRICE_STRATEGY: '0x4c707764cbFB4FFa078e169e6b8A6AdbE7526a2c',
```

File: scripts/configs/networks/ethereum.ts (hunk 5 above)
Comment (parth-15) [reply scripts/configs/networks/ethereum.ts]: done

File: scripts/configs/networks/ethereum.ts (hunk 6 above)
Comment (parth-15) [reply scripts/configs/networks/ethereum.ts]: done

File: scripts/configs/networks/ethereum.ts [hunk 7]
```diff
@@ -32,9 +32,15 @@ export const ethereumAddresses: NetworkAddresses<{
     GHO_TOKEN: '0x40D16FC0246aD3160Ccc09B8D0D3A2cD28aE6C2f',
     GHO_FLASHMINTER_FACILITATOR: '0xb639D208Bcf0589D54FaC24E655C79EC529762B8',
     // TODO: perhaps makes sense to create a generator for GSM to automatically pull GSMs, for now let's see how this evolves
-    GSM_REGISTRY: '0x0d8eFfC11dF3F229AA1EA0509BC9DFa632A13578',
+    GSM_REGISTRY: '0x167527DB01325408696326e3580cd8e55D99Dc1A',
     GSM_USDC: '0x0d8eFfC11dF3F229AA1EA0509BC9DFa632A13578',
     GSM_USDT: '0x686F8D21520f4ecEc7ba577be08354F4d1EB8262',
+    GSM_USDC_FIXED_FEE_STRATEGY: '0xD4478A76aCeA81D3768A0ACB6e38f25eEB6Eb1B5',
+    GSM_USDT_FIXED_FEE_STRATEGY: '0xD4478A76aCeA81D3768A0ACB6e38f25eEB6Eb1B5',
```
Comment (miguelmtzinf) [inline scripts/configs/networks/ethereum.ts]:
```suggestion
    GSM_USDC_FEE_STRATEGY: '0xD4478A76aCeA81D3768A0ACB6e38f25eEB6Eb1B5',
    GSM_USDT_FEE_STRATEGY: '0xD4478A76aCeA81D3768A0ACB6e38f25eEB6Eb1B5',
```

**Review by miguelmtzinf — APPROVED** [review]

**Pull request closed without merging**